### DISCUSSION TITLE
How to transfer custom data of crossing process？

### DISCUSSION THREAD

**Discussion by YuWan1117** [apache/skywalking#8558] [Q&A] [answered]
###  Header Protocol V1.0
According to the document, I get the way of [SkyWalking Cross Process Correlation Headers Protocol V1.0](https://github.com/apache/skywalking/blob/master/docs/en/protocols/Skywalking-Cross-Process-Correlation-Headers-Protocol-v1.md).
The "sw8-correlation" headerName can transfer custom data, but it has some disadvantage that the custom KV information is in value of "sw8-correlation" header and is encoded by base64.

###  Header Protocol V3.0
According to the document, I also get the way of [SkyWalking Cross Process Propagation Headers Protocol V 3.0](https://github.com/apache/skywalking/blob/master/docs/en/protocols/Skywalking-Cross-Process-Propagation-Headers-Protocol-v3.md).
Bu the "sw8-x" header it mentions is still not the same as I expected. The x (of the sw8-x) is not a key that supports customization. And I have not  understand the scenarios for using sw8-x header.

## My Requirement
> KV information such as `user = 12345` , then header info is `sw8-user = 12345`.

**Comment by YuWan1117** [comment]
Look forward to your help and guidance. 
Feel free to share your own implementation of similar scenarios, and give me some inspiration. 
Thanks!!!

**Comment by wu-sheng** [comment, marked as answer]
This is what you means https://skywalking.apache.org/docs/main/v8.9.1/en/protocols/skywalking-cross-process-correlation-headers-protocol-v1/

Just some agents implemented this.

**Reply by YuWan1117** [reply]
Thank you for your reply.

I've read the documentation you mentioned, but there are differences in my usage scenarios.
In the document, the format is `sw8-correlation = base64(string key):base64(string value),base64(string key2):base64(string value2)`.
But I want the format that is `sw8-${custom_key} = ${custom_value}`.

I don't know if it is clearly described?

**Reply by wu-sheng** [reply]
The protocol is the protocol. What you are using is not the case.

**Reply by wu-sheng** [reply]
This is an open source community. Until you are writing codes, what you want is only your own business, not ours.

**Reply by YuWan1117** [reply]
I want to confirm if a similar scenario is supported.
If not, I will see the source code to try it. 
Thanks~

**Reply by wu-sheng** [reply]
I think we are clear about what the protocol is. I don't have more to confirm

**Reply by YuWan1117** [reply]
OK
